lib/src/serialization/deserializer.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// deserializer.dart
+// A deserialization block, deserializing narrow input data onto a wide channel.
+//
+// 2024 August 27
+// Author: desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'package:collection/collection.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// [Deserializer] aggregates data from a serialized stream.
+class Deserializer extends Module {
+  /// Aggregated data output.
+  LogicArray get deserialized => output('deserialized') as LogicArray;
+
+  /// Length of aggregate to deserialize.
+  int get length => _length;
+
+  /// [done] emitted when the last element is committed to [deserialized].
+  /// The timing is that you can latch [deserialized] when [done] is high.
+  Logic get done => output('done');
+
+  /// Return the current count of elements that have been serialized out.
+  Logic get count => output('count');
+
+  /// Internal storage of length of data to be serialized.
+  late final int _length;
+
+  /// Build a Deserializer that takes serialized input [serialized]
+  /// and aggregates it into one wide output [deserialized] of length [length].
+  ///
+  /// Updates one element per clock while [enable] (if connected) is high,
+  /// emitting [done] when completing the filling of wide output `LogicArray`
+  /// [deserialized].
+  Deserializer(Logic serialized, this._length,
+      {required Logic clk,
+      required Logic reset,
+      Logic? enable,
+      super.name = 'Deserializer'}) {
+    clk = addInput('clk', clk);
+    reset = addInput('reset', reset);
+    if (enable != null) {
+      enable = addInput('enable', enable);
+    }
+    serialized = addInput('serialized', serialized, width: serialized.width);
+    final cnt = Counter.simple(
+        clk: clk, reset: reset, enable: enable, maxValue: length - 1);
+    addOutput('count', width: cnt.width) <= cnt.count;
+    addOutput('done') <= cnt.overflowed;
```

Comment:
i think this `overflowed` indication (instead of `equalsMax`) works because it happens the cycle after `equalsMax` since the data output `deserialized` is delayed by one `flop` cycle.  however, what if `enable` drops at the end of the data, then the counter would not overflow?  i'm not sure if there's a bug here or not, just wondering if it's safe to use this instead of flopping `equalsMax`?  Although similarly, if you use `equalsMax`, if enable halts at the last one then you'd hold `done` high perpetually without an edge detection.  Have you tested this case where enable drops at the end?